File: backend/app/schemas/assistant.py
from pydantic import BaseModel
from typing import Literal

from app.schemas.score import IntakeAnswers

Currency = Literal["USD", "INR"]


class ChatMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str


class AssistantChatRequest(BaseModel):
    messages: list[ChatMessage]
    intake: IntakeAnswers
    currency: Currency = "USD"


class AssistantChatResponse(BaseModel):
    reply: str
    source: Literal["command", "ollama", "fallback"] = "fallback"
    reset: bool = False


class AssistantStatusResponse(BaseModel):
    api: bool = True
    ollama: bool = False
    model: str = "llama3.2"

File: frontend/src/types/assistant.ts
import type { Currency } from "@/lib/currency";
import type { IntakeAnswers } from "@/types/intake";

export interface AssistantChatMessage {
  role: "user" | "assistant";
  content: string;
}

export interface AssistantChatRequest {
  messages: AssistantChatMessage[];
  intake: IntakeAnswers;
  currency: Currency;
}

export interface AssistantStatus {
  api: boolean;
  ollama: boolean;
  model: string;
}

export interface AssistantChatResponse {
  reply: string;
  source: "command" | "ollama" | "fallback";
  reset?: boolean;
}
